openedx/core/lib/api/view_utils.py

```diff
@@ -104,14 +105,9 @@ def _wrapper(self, request, *args, **kwargs):
                         depth=depth,
                         check_if_enrolled=True,
                     )
-                except Http404:
-                    # any_unfulfilled_milestones called a second time since has_access returns a bool
-                    if check_for_milestones and any_unfulfilled_milestones(course_id, request.user.id):
-                        message = {
-                            "developer_message": "Cannot access content with unfulfilled "
-                                                 "pre-requisites or unpassed entrance exam."
-                        }
-                        return response.Response(data=message, status=status.HTTP_204_NO_CONTENT)
+                except Http404 as error:
```

Comment:
Can you just catch "CoursewareAccessException" directly instead of catching all 404s and reraising most of them?